infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutTrackNumberService.kt

```diff
@@ -190,6 +236,36 @@ private fun asCsvFile(items: List<TrackLayoutKmLengthDetails>, translation: Tran
     return printCsv(columns, items)
 }
 
+private fun isGeneratedRow(kmPost: TrackLayoutKmLengthDetails): Boolean =
+    kmPost.layoutGeometrySource == GeometrySource.GENERATED
+
+private fun locationSourceTranslationKey(
+    kmPost: TrackLayoutKmLengthDetails,
+    precision: KmLengthsLocationPrecision,
+): String {
```

Comment:
Primitiivin voisi tyyppitarkennuksena muuttaa tässä ja alemmissa LocalizationKey-tyyppiin

Reply:
Tässä on käytössä `t()`:n on overload, joka ottaa sisäänsä stringejä ja muuntaa ne sisäisesti `LocalizationKey`:iksi, eli käytännössä tyyppiturvallisuus tulee jo varmistettua nytkin (varsinkin kun `Translation` pelaa sisäisesti `LocalizationKey`:illä.) Omasta mielestäni tämä on selkeämpi näin, kun ei ole ylimääräisiä rakenteluja luomassa noisea

Reply:
Itse taas tulkitsen että staattinen (käännösaikainen) tyypitys tarkentuisi sekä tulkinta helpottuisi, koska muuttujien tyypit olisivat heti lähdepaikassa sitä mikä niiden käyttötarkoitus onkin, tyyppitiedonhan perusidea on tuoda lisää kontekstitietoa ohjelmoijalle ja kääntäjälle. Tarkemmalla tyypityksellä muuttujien tyypit eivät myöskään vaihtuisi juuri ennen käyttöä siihen tarkennettuun tyyppiin vieläpä ns. implisiittisellä tyyppimuunnoksella, jonka helppokäyttöisyys-overloadi tekee 🤔. Hieman tulee vibat dynaamisemmasta ohjelmointikielestä tällä rakenteella (toki onhan tämä käännöshärveli muodostettu jäsärakenteen perusteellakin että ei sinänsä ihmetys).

Tuo ajonaikainen varmistus LocalizationKey-rakentajassa on sinällään omaan mieleen hieman ikävähkö jo muutenkin, joskin varmaan niihin sentään jokin automaattitesti ainakin yleensä osuu.

Reply:
Ehkä tuosta koko bäkkäripuolen käännös-APIsta olisi hyvä puhua porukalla vaikka dailyn perään maanantaina. Nykyisellään mielestäni `t()` on jonkinlaisessa välitilassa, sillä fronttipuolen vastineensa tavoin se hyväksyy kyllä joitain merkkijonoja joille ei löydy käännöstä ja palauttaa ne sellaisenaan, mutta frontista poiketen se ei kuitenkaan `LocalizationKey`-sidontansa takia hyväksy läheskeään kaikkia mahdollisia merkkijonoja. Itse haluaisin (ja olisin halunnut jo sitä alunperin tehdessäni) tehdä siitä mahdollisimman 1:1 frontin `t()`:n kanssa (yhtenäisyys kunniaan - osaat yhden kunnolla niin osaat toisenkin.) Tai jos tuo ei käy niin sitten ne pitäisi erottaa toisistaan siten, että on selvempää että kyseessä ei edes yritä olla samalla tavalla toimiva rajapinta.

Reply:
Muutin nämä nyt `LocalizationKey`:iksi. `.let()`:illä nämä sai lopulta oikeaan muotoon onneksi aika helpolla